nightfall-optimist/src/app.mjs

```diff
@@ -12,6 +16,49 @@ import {
   debug,
 } from './routes/index.mjs';
 
+const { optimistApiUrl } =
+  config.ENVIRONMENTS[process.env.ENVIRONMENT] || config.ENVIRONMENTS.localhost;
+
+const options = {
+  definition: {
+    openapi: '3.0.0',
+    info: {
+      title: 'Nightfall Optmist API',
+      version: '1.0.0',
+      description: 'An api to be used by the proposers',
+    },
+    servers: [
+      {
+        url: `${optimistApiUrl}`,
+      },
+    ],
+    components: {
+      securitySchemes: {
+        ApiKeyAuth: {
+          type: 'apiKey',
+          name: 'X-API-KEY',
+          in: 'header',
+        },
+      },
+      responses: {
+        UnauthorizedError: {
+          description: 'API key is missing or invalid',
+          headers: {
+            WWW_Authenticate: {
+              schema: {
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  apis: ['src/routes/**/*.mjs'],
+};
+
+const spec = swaggerJsDoc(options);
```

Comment:
We could try to move this outside `app.mjs` and simply export/import `spec` from the new file into this one